src/hooks/useAlgorithmRunnerWithSteps.ts

```diff
@@ -0,0 +1,8 @@
+import { useState } from "react";
+import { AlghorithmRunParams, AlgorithmRunnerWithSteps } from "../logic/algo/runnerWithSteps";
+import { initialRunParams } from "../state/runParamsReducer";
+
+export function useAlgorithmRunnerWithSteps(): Generator<unknown, unknown, unknown> {
+  const [generator, setGenerator] = useState(AlgorithmRunnerWithSteps.run(initialRunParams));
+  return generator;
+}
```

Comment:
Zostawiłam ten plik useAlgorithmRunnerWithSteps, nie wiem za bardzo jak mogłabym to wpakować do useAlgorithmRunner :(

Reply:
Myślę, ze lepiej jeśli on będzie osobno, bo raczej będzie mieć zupełnie inną logikę.

A tak swoją drogą, to mogłabyś poprawić też literówkę w nazwie pliku `useAlgorithmRunner`
i też w `AlghorithmRunParams`?